threat_model/threat_model.md

```diff
@@ -0,0 +1,99 @@
+- Malicious TF state can cause information leaks when used by TF client to deploy an application.
+- Vault keys are unique per user and not shared.
+- Certificates issued by CAs are trustworthy.
+  - If they are not, any use of "proper certificates" can be further secured with certificate pinning.
+
+
+## Use cases
+
+### UC1: Get State
+A Terraform client calls the Terraform Vault Backend to obtain previously stored TF state.
+
+```mermaid
+sequenceDiagram
+  participant TF client
+  participant TVB
+  participant Vault
+
+  TF client->>TVB: 1: GET /state/$PATH <Vault API key>
+  TVB->>Vault: 2: GET <Vault API key>
+  Vault->>TVB: 3: GET response <state>
+  TVB->>TF client: 4: GET response <state>
+```
+
+| Use case | Attacker | What | Impact | Exposure | Severity | Countermeasures | Verification |
+|----------|----------|------|--------|----------|----------|-----------------|--------------|
+| UC1:1 | UIA | DoS server with garbage traffic | Low | High | Medium | Ensure service has sufficient resources to withstand lone attackers | Load test deployment
+|||||| Low | Deploy service on internal network or behind DoS protection | Check that deployed IPs belong to known good range
+| UC1:1 | UIA | Exhaust server resources using clever tricks like slow loris | Low | High | Medium | Deploy service behind mitigating reverse proxy (nginx, caddy, etc.) | Test that slow lorising deployment fails
+|||||| Low | Deploy service on internal network or behind DoS protection | Check that deployed IPs belong to known good range
+| UC1:1 | CNA | Impersonate Vault/TVB to obtain Vault API keys | High | Medium | High | Protect traffic with TLS + proper certificates | Test that TVB deployment rejects plain HTTP and uses a non-self signed cert
+| UC1:1 | CNA | Eavesdrop on traffic to obtain Vault API keys | High | Medium | High | Enforce TLS on TVB side | Test that TVB deployment rejects plain HTTP
+| UC1:1 | UIA, MUA | Impersonate user to obtain secrets from TF state | High | High | High | Use individual Vault API keys | Audit logs for signs of shared keys
+|||||| Low | Log accesses with identity and IP | Test that identity and IP is written to log for each request
+| UC1:1 | CTA | Obtain and leak Vault API keys | High | Low | High | Ensure dependencies are minimal, trustworthy, and up to date | Use automatic dependency scanning in CI
+|||||| Medium | Deploy IDS to increase chance of detection | Test deployment for IDS presence
+| UC1:3 | CVA, CNA | Return malicious state to user which steals data from deployment if deployed | High | Medium | High | Protect traffic with TLS + proper certificates | Test that TVB refuses to talk to HTTP or self-signed Vault
+| UC1:3 | CNA | Eavesdrop on returned TF state to obtain secrets | High | Medium | High | Protect traffic with TLS | See above
+| UC1:3 | CTA | Leak secrets obtained from Vault | High | Low | High | Ensure dependencies are minimal, trustworthy, and up to date | Use automatic dependency scanning in CI
```

Comment:
Another countermeasure is to turn off general egress traffic from TVA (like we are doing).